### PR TITLE
docs: add public-api.md

### PR DESCRIPTION
**Pull request by lunarwhite**
### Description

add document under `docs/zh/latest/plugins/`

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: docs/zh/latest/plugins/public-api.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+#
+-->
+
+## 描述
+
+`public-api` 插件用于通过一个通用的 HTTP API 路由暴露一个 API 端点。
+
+当你使用自定义插件时，你可以使用 `public-api` 插件为特定功能定义一个固定的公共 API。例如，你可以使用 [`jwt-auth`](./jwt-auth.md) 插件创建一个公共 API 端点 `/apisix/plugin/jwt/sign` 用于 JWT 认证。
+
+默认情况下，在自定义插件中添加的公共 API 是不公开的，用户需要手动配置一个路由并在上面启用 `public-api` 插件。
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/public-api.md:36]:
```suggestion
:::note 注意

默认情况下，在自定义插件中添加的公共 API 是不公开的，用户需要手动配置一个路由并在上面启用 `public-api` 插件。

:::
```

File: docs/zh/latest/plugins/public-api.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+## 属性
+
+| 名称  | 类型   | 必选项    | 默认值   | 描述                                                        |
+|------|--------|----------|---------|------------------------------------------------------------|
+| uri  | string | 否       | ""      | 公共 API 的 URI。在设置路由时，使用此属性来配置初始的公共 API URI。 |
+
+## 启用插件
+
+除了 `public-api` 插件，下面的例子也使用了 [`jwt-auth`](./jwt-auth.md) 和 [`key-auth`](./key-auth.md) 插件，详细使用方法请参考它们对应的文档。
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/public-api.md:46]:
```suggestion
除了 `public-api` 插件，下面的例子也使用了 [`jwt-auth`](./jwt-auth.md) 和 [`key-auth`](./key-auth.md) 插件，详细使用方法请参考各自对应的文档。
```

File: docs/zh/latest/plugins/public-api.md (hunk 1 above)
Comment (hf400159) [reply docs/zh/latest/plugins/public-api.md:36]: 默认情况下，在自定义插件中添加的公共 API 不对外暴露的，你需要手动配置一个路由并启用 `public-api` 插件。

File: docs/zh/latest/plugins/public-api.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    -H 'Content-Type: application/json' \
+    -d '{
+    "uri": "/apisix/plugin/jwt/sign",
+    "plugins": {
+        "public-api": {}
+    }
+}'
+```
+
+**测试插件**
```
Comment (hf400159) [inline docs/zh/latest/plugins/public-api.md:64]: Should a consumer need to be created? Otherwise, the test cannot be performed.

**Review by SylviaBABY — DISMISSED** [review]

File: docs/zh/latest/plugins/public-api.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+## 描述
+
+`public-api` 插件可用于通过创建路由的方式暴露用户自定义的 API。
+
+你可以通过在路由中添加 `public-api` 插件，来保护**自定义插件为了实现特定功能**而暴露的 API。例如，你可以使用 [`jwt-auth`](./jwt-auth.md) 插件创建一个公共 API 端点 `/apisix/plugin/jwt/sign` 用于 JWT 认证。
+
+:::note 注意
+
+默认情况下，在自定义插件中添加的公共 API 不对外暴露的，你需要手动配置一个路由并启用 public-api 插件。
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/public-api.md:37]:
```suggestion
默认情况下，在自定义插件中添加的公共 API 不对外暴露的，你需要手动配置一个路由并启用 `public-api` 插件。
```

**Review by SylviaBABY — APPROVED** [review]

**Review by SkyeYoung — APPROVED** [review]

**Review by guoqqqi — APPROVED** [review]

**Pull request merged**